### DISCUSSION TITLE
Get commit hash link of a file

### DISCUSSION THREAD

**Discussion by 134dd3v** [community/community#17668] [Other Feature Feedback, Questions, & Ideas] [answered]
Currently when browsing files on GitHub, when we want to share link to some code snippet, it is considered best practice to instead share the commit hash link, instead of the branch head. But currently there doesn't seem to be an easy way to get that link if one is already on the file page. After clicking commit hash -> Browse Files, it takes to the repository root. It'd be great if there would be some way to go to the file with commit hash link if one is on the file page.

**Comment by laymonage** [comment, marked as answer]
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet

https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files

> When viewing a file on GitHub.com, you can press the "y" key to update the URL to a permalink to the exact version of the file you see.

<img width="1266" alt="image" src="https://user-images.githubusercontent.com/6379424/171770289-80ed617c-9e15-476a-8008-4fa6f2def067.png">

![image](https://user-images.githubusercontent.com/6379424/171770358-b53847cd-d3b9-42d9-95e4-032828cf7197.png)

**Reply by 134dd3v** [reply]
Thank you!
